pyproximal/proximal/ETP.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+from scipy.special import lambertw
+
+from pyproximal.ProxOperator import _check_tau
+from pyproximal import ProxOperator
+
+
+class ETP(ProxOperator):
+    r"""Exponential-type penalty (ETP).
+
+    Parameters
+    ----------
+    sigma : :obj:`float`
+        Regularization parameter.
+    gamma : :obj:`float`, optional
+        Regularization parameter. Default is 1.0.
+
+    Notes
+    -----
+    The exponential-type penalty is defined as
+
+    .. math::
+
+        ETP_{\sigma,\gamma}(\mathbf{x}) = \sum_i \frac{\sigma}{1-e^{-\gamma}}(1-e^{-\gamma|x_i|})
+
+    for :math:`{\sigma>0}`, and :math:`{\gamma\geq 0}`. Note that when
+    :math:`{\gamma\rightarrow 0}` the logarithmic penalty approaches the l1-penalty and when
```

Comment:
Do you mean the exponential when you say logarithmic

Reply:
Yes, I will fix it in an update.

Reply:
I can see if it possible to vectorize this function, it depends on the lambertw function a bit, but it says that it supports array_like in the documentation of scipy.

Reply:
Alright, have a look, that would be great! If not, or not so obvious, we can leave it as is for now and make an issue that reminds us perhaps we can get a vectored version in future :)

Reply:
It wasn't too hard.